gateway_selector/public/js/base.js

```diff
@@ -196,8 +193,55 @@ frappe.integration_service.gateway_selector_gateway = Class.extend({
 
   process: function(overrides, callback) {
     if ( this.current_gateway ) {
-      this.current_gateway.collect();
-      this.current_gateway.process(overrides, callback);
+      let processCall = () => {
+        this.current_gateway.collect();
+        this.current_gateway.process(overrides, callback);
+      };
+
+      // fuggly way of updating gateway proxy record with gateway selection.
+      // we first update the record, wait, then do the actual processing.
+      if ( this.request_data.proxy_name ) {
+        frappe.call({
```

Comment:
Just a suggestion: you could use `frappe.db.set_value` on the client-side, which is just a `frappe.call` to `set_value` on the server-side. It'll make this slightly more readable and you won't need the whitelist method on the py.

Reply:
So, this page is served from the front end. None of the frappe.db apis are available there.
Maybe a better route would be calling frappe.client.set_value directly but it wouldn't really shrink this masterpiece